src/test/kotlin/io/inventi/eventstore/eventhandler/util/WithMeterRegistryAssertions.kt

```diff
@@ -0,0 +1,29 @@
+package io.inventi.eventstore.eventhandler.util
+
+import io.micrometer.core.instrument.Gauge
+import io.micrometer.core.instrument.MeterRegistry
+import org.junit.Assert.assertEquals
+import org.testcontainers.shaded.org.awaitility.Awaitility
+import java.util.concurrent.TimeUnit
+
+interface WithMeterRegistryAssertions {
+    var meterRegistry: MeterRegistry
+
+    private val assertDelta: Double
+        get() = 1e-6
+
+    fun assertGaugeWithTagHasValue(gaugeName: String, meterTag: String, meterValue: Double) {
+        Awaitility.await().pollDelay(1, TimeUnit.SECONDS).atMost(5, TimeUnit.SECONDS).untilAsserted {
+            assertEquals(valueOfGaugeWithTag(gaugeName, meterTag), meterValue, assertDelta)
+        }
+    }
+
+    private fun valueOfGaugeWithTag(gaugeName: String, meterTag: String): Double {
+        return meterRegistry.find(gaugeName)
```

Comment:
Meter registry has api to look for tags:
```
return meterRegistry.find(gaugeName)
    .tag(tagName, tagValue)
    .gauge()!!
    .value()
```

Reply:
Nice, done